### PR TITLE
JW-1242: vRouter cleanup, 5th patchset

### PR DESCRIPTION
**Pull request by sodar**
- reformatting of NBL processing code

### CODE REVIEW

File: windows/vr_nbl.c [hunk 1]
```diff
@@ -10,108 +10,100 @@ FILTER_SEND_NET_BUFFER_LISTS FilterSendNetBufferLists;
...
 PNET_BUFFER_LIST
-create_nbl(unsigned int size)
+CreateNetBufferList(unsigned int BytesCount)
 {
-    ASSERT(size > 0);
-
-    PNET_BUFFER_LIST nbl = NULL;
-    void *buffer = NULL;
+    if (BytesCount == 0)
+        return NULL;
```
Comment (michal-clapinski) [inline windows/vr_nbl.c]: leave the assert

File: windows/vr_nbl.c [hunk 2]
```diff
@@ -10,108 +10,100 @@ FILTER_SEND_NET_BUFFER_LISTS FilterSendNetBufferLists;
...
-fail:
-    if (buffer)
-        ExFreePoolWithTag(buffer, VrAllocationTag);
-    return NULL;
 }
 
-void
-free_cloned_nbl(PNET_BUFFER_LIST nbl)
+VOID
+FreeClonedNetBufferList(PNET_BUFFER_LIST Nbl)
```
Comment (michal-clapinski) [inline windows/vr_nbl.c]: stick to `nbl`

File: windows/vr_nbl.c [hunk 3]
```diff
@@ -120,45 +112,45 @@ free_created_nbl(PNET_BUFFER_LIST nbl)
...
             else
-                free_created_nbl(nbl);
+                FreeCreatedNetBufferList(Nbl);
 
-            PNET_BUFFER_LIST parent = nbl->ParentNetBufferList;
-            if (parent)
-                free_nbl(parent);
+            if (parent != NULL)
+                FreeNetBufferList(parent);
+        } else {
```
Comment (michal-clapinski) [inline windows/vr_nbl.c]: you mix using brackets with not using them

File: windows/vr_nbl.c [hunk 4]
```diff
@@ -185,40 +177,37 @@ delete_unbound_nbl(PNET_BUFFER_LIST nbl, unsigned long flags)
...
-    if (nbl == NULL)
+    BOOLEAN contextCreated = false;
+    PNET_BUFFER_LIST newNbl = NdisAllocateCloneNetBufferList(OriginalNbl, VrNBLPool, NULL, 0);
+    if (newNbl == NULL)
         goto cleanup;
 
-    nbl->SourceHandle = VrSwitchObject->NdisFilterHandle;
-    nbl->ParentNetBufferList = original_nbl;
-    original_nbl->ChildRefCount++;
+    newNbl->SourceHandle = VrSwitchObject->NdisFilterHandle;
```
Comment (michal-clapinski) [inline windows/vr_nbl.c]: you generally mix variables' names starting with big letter and starting with small letter

**Comment by sodar** [comment]
Retest this please

File: windows/vr_nbl.c [hunk 5]
```diff
@@ -10,105 +10,99 @@ FILTER_SEND_NET_BUFFER_LISTS FilterSendNetBufferLists;
...
-    return VrSwitchObject->NdisSwitchHandlers.AllocateNetBufferListForwardingContext(VrSwitchObject->NdisSwitchContext, nbl);
+    ASSERT(Nbl != NULL);
+    return VrSwitchObject->NdisSwitchHandlers.AllocateNetBufferListForwardingContext(
+        VrSwitchObject->NdisSwitchContext,
+        Nbl);
 }
 
 static void
-free_forwarding_context(PNET_BUFFER_LIST nbl)
+FreeForwardingContext(PNET_BUFFER_LIST Nbl)
```
Comment (michal-clapinski) [inline windows/vr_nbl.c]: still mixing `Nbl` with `nbl`, grep it or something

File: windows/vr_nbl.c (hunk 4 above)
Comment (michal-clapinski) [reply windows/vr_nbl.c]: seriously, you should decide on one

Comment (michal-clapinski) [reply windows/vr_nbl.c]: I prefer small-case in case you were wondering

File: windows/vr_nbl.c [hunk 6]
```diff
@@ -234,9 +225,9 @@ win_allocate_packet(void *buffer, unsigned int size)
     unsigned char *ptr = NULL;
 
     if (buffer != NULL) {
-        nbl = create_nbl_based_on_buffer(size, buffer);
+        nbl = CreateNetBufferListUsingBuffer(size, buffer);
     } else {
-        nbl = create_nbl(size);
+        nbl = CreateNetBufferList(size);
```
Comment (michal-clapinski) [inline windows/vr_nbl.c]: In this `if` there are braces...

File: windows/vr_nbl.c [hunk 7]
```diff
@@ -257,20 +248,19 @@ win_allocate_packet(void *buffer, unsigned int size)
     if (pkt)
         NdisFreeNetBufferListContext(nbl, VR_NBL_CONTEXT_SIZE);
```
Comment (michal-clapinski) [inline windows/vr_nbl.c]: and in this one there are none (but this doesn't have to be fixed in this review)

File: windows/vr_host.c [hunk 8]
```diff
@@ -345,7 +345,7 @@ win_pclone(struct vr_packet *pkt)
         goto cleanup_nbl;
     *npkt = *pkt;
 
-    pkt->vp_ref_cnt++;
+    vr_sync_add_and_fetch_32u(&pkt->vp_ref_cnt, 1);
```
Comment (michal-clapinski) [inline windows/vr_host.c]: woah, why is that?

File: windows/vr_nbl.c [hunk 9]
```diff
@@ -10,105 +10,101 @@ FILTER_SEND_NET_BUFFER_LISTS FilterSendNetBufferLists;
...
 
 static NDIS_STATUS
-create_forwarding_context(PNET_BUFFER_LIST nbl)
+CreateForwardingContext(PNET_BUFFER_LIST nbl)
 {
     ASSERT(nbl != NULL);
-    return VrSwitchObject->NdisSwitchHandlers.AllocateNetBufferListForwardingContext(VrSwitchObject->NdisSwitchContext, nbl);
+    return VrSwitchObject->NdisSwitchHandlers.AllocateNetBufferListForwardingContext(
+        VrSwitchObject->NdisSwitchContext,
+        nbl);
```
Comment (michal-clapinski) [inline windows/vr_nbl.c]: I'd merge line 17 and 18.

File: windows/vr_host.c (hunk 8 above)
Comment (sodar) [reply windows/vr_host.c]: There might be cases where packet and its clones might access parent's `vp_ref_cnt` simultaneously. I have briefly discussed this with @jablonskim and we came to the conclusion that incrementations and decrementations should be synchronized.

File: windows/vr_nbl.c (hunk 9 above)
Comment (sodar) [reply windows/vr_nbl.c]: I will do this in the next PR.

**Pull request merged**